onnxscript/rewriter/generic_pattern.py

```diff
@@ -297,20 +299,33 @@ def _match_backward(
             )
             return self.none(starting_node, inspect.currentframe().f_lineno)
 
-        for graph_input, pattern_input in zip(graph_node.inputs, pattern_node.inputs):
-            if len(list(graph_input.uses())) != len(list(pattern_input.uses())):
-                self._hint(
-                    "BACKWARD: one input is used outside the pattern",
-                    "-- pattern",
-                    pattern_node,
-                    "-- model",
-                    graph_node,
-                )
-                return self.none(starting_node, inspect.currentframe().f_lineno)
+        has_predecessor_in_pattern = any(
+            pattern_value.producer() is not None for pattern_value in pattern_node.inputs
+        )
+
+        if has_predecessor_in_pattern:
```

Comment:
Is this condition meant to be an optimization (with no effect in behavior)? Trying to understand

Reply:
I think this fix might be related to an issue Jingyan is running into. Does this fix allow pattern-inputs to have other uses outside of the pattern?

Reply:
I think so. I think we should change this slightly, by applying the condition for each pattern_value separately, as mentioned in comment below.

Reply:
Fill free to rewrite the code. If the unit test are passing, it means it is good.